pkgs/development/ocaml-modules/opti/default.nix

```diff
@@ -0,0 +1,18 @@
+{ stdenv, fetchurl, buildDunePackage }:
+
+buildDunePackage rec {
+  pname = "opti";
+  version = "1.0.3";
+
+  src = fetchurl {
```

Comment:
please use `fetchFromGitHub` instead of fetchurl

Reply:
No, fetchurl is valid here. This is fetching a Github release tarball. fetchFromGitHub is for fetching the autogenerated tarball for a Git tag, it doesn't support fetching a release tarball.

Reply:
I've checked the difference between both sources and I agree with @catern here as the released tarball has the version defined in it.

```path
diff -Naur not-generated/.gitignore generated/.gitignore
--- not-generated/.gitignore    1969-12-31 16:00:00.000000000 -0800
+++ generated/.gitignore        2018-11-30 11:35:52.149183398 -0800
@@ -0,0 +1,27 @@
+*.annot
+*.cmo
+*.cma
+*.cmi
+*.a
+*.o
+*.cmx
+*.cmxs
+*.cmxa
+
+# ocamlbuild working directory
+_build/
+
+# ocamlbuild targets
+*.byte
+*.native
+
+# oasis generated files
+setup.data
+setup.log
+
+* emacs generated backups
+*~
+
+# dune generated files
+*.merlin
+*.install
diff -Naur not-generated/opti.opam generated/opti.opam
--- not-generated/opti.opam     2018-11-25 00:08:19.000000000 -0800
+++ generated/opti.opam 2018-11-30 11:35:52.151183390 -0800
@@ -1,4 +1,3 @@
-version: "1.0.3"
 opam-version: "2.0"
 synopsis: "DSL to generate fast incremental C code from declarative specifications"
 description: "Generate C code to incrementally compute sums, products and other expressions, without recomputing them from scratch every time."
@@ -16,4 +15,4 @@
 build: [
   ["dune" "build" "-p" "opti" "-j" jobs]
   ["dune" "runtest" "-p" "opti" "-j" jobs] {with-test}
-]
\ No newline at end of file
+]
```